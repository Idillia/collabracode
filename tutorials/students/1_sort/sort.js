
function addEvent(obj, evType, fn) {
  // taken from http://www.onlinetools.org/articles/unobtrusivejavascript/chapter4.html
  if (obj.addEventListener) {
    obj.addEventListener(evType, fn, false); 
    return true; 
  } else if (obj.attachEvent) {
    var r = obj.attachEvent("on"+evType, fn); 
    return r; 
  } else {
    return false; 
  }
}

function runSortOnce() {
  if(!itersort) {
    return true;
  }
  var table = document.getElementById("sort-table");
  var tbody = table.tBodies[0];
  var lastRow = tbody.rows[tbody.rows.length-1];
  var cells = lastRow.cells;
  var arr = [];
  var arr2 = [];
  for(var i=0; i<cells.length; i++) {
    var value = parseInt(cells[i].innerText);
    arr.push(value);
    arr2.push(value);
  }
  var outarr = itersort(arr);
  if(outarr.length == arr2.length) {
    var isSame = true;
    for(var i=0; i<arr2.length; i++) {
      if(outarr[i] != arr2[i]) {
        isSame = false;
        break;
      }
    }
    if(isSame) {
      return true;
    }
  }
  var newRow = document.createElement("tr");
  newRow.className = "attempt";
  for(var i=0; i<outarr.length; i++) {
    var newCell = document.createElement("td");
    newCell.className = "element";
    newCell.innerText = new String(outarr[i]);
    newRow.appendChild(newCell);
  }
  tbody.appendChild(newRow);
  return false;
}

function runSort() {
  while(!runSortOnce()) {};
}

function attachButtonEvents() {
  addEvent(document.getElementById("run-sort-button"), "click", runSort);
}

addEvent(window, "load", attachButtonEvents);

