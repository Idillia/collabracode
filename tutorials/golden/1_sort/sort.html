<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html >
<html>
  <head>
    <title>Lesson 1 -- Basic JavaScript</title>
    <script type="text/javascript" src="student_sort.js"></script>
    <script type="text/javascript" src="sort.js"></script>
    <link rel="stylesheet" href="sort.css"/>
  </head>
  <body>
    <h1>Sorting</h1>
    <div id="run-sort-button">Click to Run Sort</div>
    <table id="sort-table">
      <tbody>
        <tr class="attempt">
          <td class="element">23</td>
          <td class="element">17</td>
          <td class="element">53</td>
          <td class="element">29</td>
          <td class="element">47</td>
          <td class="element">19</td>
          <td class="element">91</td>
          <td class="element">73</td>
          <td class="element">71</td>
          <td class="element">5</td>
          <td class="element">13</td>
        </tr>
      </tbody>
    </table>
    <div>
      <div>
        To make this work, you must create a javascript file with at least the following functions:
      </div>
      <ul>
        <li><b>itersort</b> -- takes in an array and outputs an array</li>
      </ul>
    </div>
  </body>
</html>
